webapp/sources/rudder/rudder-core/src/main/scala/com/normation/rudder/services/reports/ComputeNodeStatusReportService.scala

```diff
@@ -391,18 +414,25 @@ class ComputeNodeStatusReportServiceImpl(
    * Find in cache all outdated compliance, and add to queue to recompute them
    */
   override def outDatedCompliance(now: DateTime): IOResult[Unit] = {
+    import RunAnalysisKind.*
     nsrRepo.getAll()(QueryContext.systemQC).flatMap { cache =>
       val nodeWithOutdatedCompliance = cache.filter {
         case (id, compliance) =>
           compliance.runInfo.kind match {
-            // here, we have a special case for unexpected version: it is useless to recompute compliance until we don't have a new run,
-            // ie the node config was changed elsewhere. It means that "unexpected version" wins above "No report in interval",
-            // ie that that error is bigger.
-            case RunAnalysisKind.UnexpectedVersion => false
-            // other expiring status
-            case _                                 =>
+            // here, we only want to check for compliance that expires when time pass and we
+            // were in a good state. Bad states need an external action (new config, new runs)
+            // to change back to good state
```

Comment:
this is the main loop-breaker. 
Now that things are retrieved from base as a cache, we really just have two kind of things : 
- the one that change just with time - and unfortunatelly, things only get worse with time, it doesn't magically repair a bad case, 
- things that need an external action (new config, new run) to change. 
Here, we only care for the first one, because we want to check that we didn't let things rotting in base. 
Also, I enumerated all cases, because it will be a point of attention to have if we ever had a new status, I don't want to let it be caught by a "*" where it doesn't belong.